index.js

```diff
@@ -1,85 +1,157 @@
-const xtend = require('xtend')
 const EventEmitter = require('events').EventEmitter
-const sodium = require('sodium-universal')
-const bufferAlloc = require('buffer-alloc-unsafe')
 const parallel = require('run-parallel')
 const encoding = require('dat-encoding')
+const RAW = require('random-access-web')
+const RAM = require('random-access-memory')
+const DiscoverySwarmWeb = require('discovery-swarm-web')
+const crypto = require('hypercore-crypto')
+const hypercoreProtocol = require('hypercore-protocol')
+const hyperdrive = require('hyperdrive')
 
-const Repo = require('./repo')
+const DAT_PROTOCOL = 'dat://'
+const STORAGE_NAME = 'dats'
+const STORAGE_COLLECTION_NAME = `files`
 
 module.exports =
 
 /**
- * The Dat object. Manages multiple repositories in
- * a single discovery-swarm instance.
+ * The Dat object. Manages multiple archives in
+ * a single discovery-swarm-web instance.
  * @param {Object} opts   Default options to use for the dat.
  */
 class Dat extends EventEmitter {
   constructor (opts) {
     super()
     this.opts = opts || {}
 
-    if (!this.opts.id) this.opts.id = randomBytes(32)
+    if (!this.opts.id) this.opts.id = crypto.randomBytes(32)
 
-    this.repos = []
+    this.archives = []
+
+    const discoveryOpts = Object.assign({
+      stream: (info) => this._replicate(info)
+    }, this.opts)
+    this.swarm = new DiscoverySwarmWeb(discoveryOpts)
+
+    const storageOpts = Object.assign({
+      name: STORAGE_NAME,
+      storeName: STORAGE_COLLECTION_NAME
+    }, this.opts)
+    this.persistence = RAW(storageOpts)
```

Comment:
What options are being passed on? RAW doesn't have any options documented, so not clear now these are being used.

Reply:
I'll add to the docs, but `name` is the only one that does something across different storage layers.

Reply:
I added docs on the options: https://www.npmjs.com/package/random-access-web#options